tests/system/marathon_common_tests.py

```diff
@@ -1154,3 +1157,173 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(timeout=timedelta(minutes=1).total_seconds(), app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region (it's ok if they belong on different zones)
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(local_domain.region,))
+
+    # Remove app
+    client.remove_app(app['id'], True)
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+def test_faultdomains_region_only():
+    """Tests if the applications that only have a `region` specified are launched on all the different
+       zones available in that region on the cluster
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region (%s)" % (
+        region,
+    )
+
+    # Create an app pinned on the given region, with at least double the number of instances than
+    # the zones in the cluster, to ensure that they are distributed among all of them
+    app_def = apps.faultdomain_app(region=region, instances=len(zones) * 2)
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(timeout=timedelta(minutes=1).total_seconds(), app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=zones)
+
+    # Remove app
+    client.remove_app(app['id'], True)
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+def test_faultdomains_region_and_zone():
+    """Tests if the applications with a `region` and a `zone` defined are always starting on the
+       same region and zone
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region (%s)" % (
+        region,
+    )
+
+    single_zone = zones[0]
+
+    # Create an app pinned on the given region and zone.
+    #
+    # (We are starting more instances than the number of zones in the region in order to increase
+    #  the chance of a faulty instance to appear on a different zone)
+    #
+    app_def = apps.faultdomain_app(region=region, zone=single_zone, instances=len(zones) * 2)
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(timeout=timedelta(minutes=1).total_seconds(), app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=(single_zone,))
```

Comment:
What does `(region,)` mean?

Reply:
It's a tuple of one item. Like `[region]`, but lighter.